### ISSUE TITLE
preset changes for AU and VST instruments on OSX via program changes

### ISSUE THREAD

**Issue by lokkiikkol** [falkTX/Carla#1337]
program changes do not change presets on the synths i tried.

AU: Vital and Helm
VST: Vital and Helm

Helm does actually show a preset dropdown in the rack view, but if i send program changes to it nothing happens. do i need to specify a specific midi channel? i.e. 1?

**Comment by moonshaadow** [comment]
It seems that both plugins don't support Program Changes.
https://github.com/mtytel/helm/issues/175

If a plugin support PC messages, I think it should work in Carla.

**Comment by lokkiikkol** [comment]
Ok. Is there any other way via savestate in Carla? I thought the whole
point of the "Map Program Changes" Setting was to allow preset changes on
plugins that don't support program changes.

**Comment by lokkiikkol** [comment]
actually, on the link you sent it states that program changes do indeed work on helm but need additional bank messages to be sent as well...

**Comment by moonshaadow** [comment]
In the Edit window, set the midi filters "map program changes". I've just test that on Helm vst. But I use linux version (Carla version 2.3.0-alpha2), I don't know if this is already implemented on OSX.
![Capture d’écran_2021-01-26_12-26-33](https://user-images.githubusercontent.com/29754932/105839422-bf774380-5fd1-11eb-8cbe-2e43c2aeae30.png)

**Comment by lokkiikkol** [comment]
you are right! i got it to work on helm too, thanks!! so it seems to be a vital problem.
